source/euler_aeos/hyperbolic_system.h

```diff
@@ -1253,69 +1286,136 @@ namespace ryujin
     DEAL_II_ALWAYS_INLINE inline auto
     HyperbolicSystemView<dim, Number>::prescribe_riemann_characteristic(
         const state_type &U,
+        const Number &p,
         const state_type &U_bar,
+        const Number &p_bar,
         const dealii::Tensor<1, dim, Number> &normal) const -> state_type
     {
-      __builtin_trap(); // untested and likely needs to be refactored
-
       static_assert(component == 1 || component == 2,
                     "component has to be 1 or 2");
 
+      const auto b = Number(eos_interpolation_b());
+      const auto pinf = Number(eos_interpolation_pinfty());
+      const auto q = Number(eos_interpolation_q());
+
+      /*
+       * The "four" Riemann characteristics are formed under the assumption
+       * of a locally isentropic flow. For this, we first transform both
+       * states into {rho, vn, vperp, gamma, a}, where we use the NASG EOS
+       * interpolation to derive a surrogate gamma and speed of sound a.
+       */
+
       const auto m = momentum(U);
       const auto rho = density(U);
       const auto vn = m * normal / rho;
 
-      const auto p = surrogate_pressure(U); // FIXME: discuss
       const auto gamma = surrogate_gamma(U, p);
-      const auto interpolation_b = ScalarNumber(eos_interpolation_b());
-      const auto x = Number(1.) - interpolation_b * rho;
-      const auto a = std::sqrt(gamma * p / (rho * x)); // local speed of sound
-
+      const auto a = surrogate_speed_of_sound(U, gamma);
 
       const auto m_bar = momentum(U_bar);
       const auto rho_bar = density(U_bar);
       const auto vn_bar = m_bar * normal / rho_bar;
 
-      const auto p_bar = surrogate_pressure(U_bar); // FIXME: discuss
       const auto gamma_bar = surrogate_gamma(U_bar, p_bar);
-      const auto x_bar = Number(1.) - interpolation_b * rho_bar;
-      const auto a_bar = std::sqrt(gamma_bar * p_bar / (rho_bar * x_bar));
+      const auto a_bar = surrogate_speed_of_sound(U_bar, gamma_bar);
 
-      /* First Riemann characteristic: v* n - 2 / (gamma - 1) * a */
+      /*
+       * Now compute the Riemann characteristics {R_1, R_2, vperp, s}:
+       *   R_1 = v* n - 2 / (gamma - 1) * a
```

Comment:
Apologies in advance if I missed this... but are we missing a covolume term here? That is, should this be:
`R_1 = v* n - 2 / (gamma - 1) * a * (1 - b * rho)` ?

Reply:
@ejtovar Good point - let's use the NASG Riemann characteristics consistently. I have updated the PR accordingly (with a similar "explicit/implicit" trick to avoid solving the nonlinear system).